analysis/source/lib/pyHelper.py

```diff
@@ -36,18 +39,26 @@ def run_script(script, folder, absolute_path = os.getcwd(), program = "python",
             print('C:/Program Files/Stata17/StataMP-64 on WIndows')
             time.sleep(20)
 
-
-
-
-    elif program == "Rscript" or program == "python": 
-        command = [program, full_path]
     elif program == 'matlab':
         command = program + f" -batch run('{full_path}')"
 
-    tic = timer()
-    p = subprocess.run(command, capture_output = True) 
-    toc = timer()
-    
+
+    if  program != "pdflatex":
+        tic = timer()
+        p = subprocess.run(command, capture_output = True) 
+        toc = timer()
+    else: #traverse to products folder
+        tic = timer()
+        full_path = os.path.join(products_loc, folder, script)
+        os.chdir(os.path.join(products_loc, folder))
+        command = program +' '+ script
+        p = subprocess.run(command)
+        p = subprocess.run('bibtex ' + script.strip('.tex')) # .tex file without extension 
+        p = subprocess.run(command) 
```

Comment:
@SimonFreyaldenhoven Why do we have several `p = subprocess.run(command)` lines?

Reply:
@kobleary this is needed to get all the internal references right. (e.g. the citations and the numbers in things like Figure 2). It usually happens automatically in the background in your latex distribution.

Without any type of \ref{} command, and no citations, you would only need to do it once.

Reply:
@SimonFreyaldenhoven ok got it! thanks